daemon/cluster/secrets.go

```diff
@@ -1,14 +1,60 @@
 package cluster
 
 import (
+	"fmt"
+	"strings"
+
 	apitypes "github.com/docker/docker/api/types"
 	types "github.com/docker/docker/api/types/swarm"
 	"github.com/docker/docker/daemon/cluster/convert"
 	swarmapi "github.com/docker/swarmkit/api"
+	"golang.org/x/net/context"
 )
 
+func getSecretByNameOrIDPrefix(ctx context.Context, state *nodeState, nameOrIDPrefix string) (*swarmapi.Secret, error) {
+	r, err := state.controlClient.ListSecrets(ctx, &swarmapi.ListSecretsRequest{
```

Comment:
I think this is an optimization, correct? Now that we run this on the daemon, is this still needed?

I don't fully know how the ListSecrets filters work; is it possible that that filter hides a secret, that would otherwise be successfully matched here?